meson.build

```diff
@@ -0,0 +1,78 @@
+project('xrayutilities', 'c',
+  version : run_command('python', '-c',
+    '''
+    try:
+        import tomllib
+    except ImportError:
+        import tomli as tomllib
+    with open("pyproject.toml", "rb") as f:
+        print(tomllib.load(f)["project"]["version"])
+    '''
+  ).stdout().strip()
```

Comment:
Note that it is possible to specify the version here as `version: '1.7.10'` without any run_command(), and then specify it in pyproject.toml as `dynamic = ["version"]`.

meson-python will then derive it from the version in meson.build. See https://packaging.python.org/en/latest/discussions/single-source-version/ for a general discussion. There are pros and cons of each approach.

Reply:
I now use `meson.build` as the only place to specify the version (hardcoded). I am not sure if this will work for all my use cases. In `.azure/azure-buildwheels.yml` I will need to find a way to assign add a revision number (or similar) to let cibuildwheel succeed with a unique version.

Reply:
It's not a huge deal either way. But FWIW you can change meson.build dynamically, both in a source checkout and as part of creating an sdist.

For setting the version interactively/in a script: https://mesonbuild.com/Rewriter.html#setting-the-project-version

For the sdist case: https://mesonbuild.com/Creating-releases.html#cement-a-version-obtained-from-vcs

Reply:
the meson rewrite seems very useful and I will adapt it in the azure pipeline.